lib/src/metta/runner/stdlib/stdlib.metta

```diff
@@ -667,6 +667,34 @@
    (let $c1 (collapse $arg1) (let $c2 (collapse $arg2)
      (let $u (subtraction-atom $c1 $c2) (superpose $u)))))
 
+(@doc assert
+  (@desc "Asserts that the result of evaluating an expression is True")
+  (@params (
+    (@param "Expression to evaluate")))
+  (@return "Unit atom if the expression evaluates to True, error - otherwise"))
+(: assert (-> Expression Atom))
+(= (assert $arg)
+   (let $atom (eval $arg)
+        (if (== $atom True)
+            ()
+            (Error $arg (Expected True, evaluated to  $atom)))))
```

Comment:
I've never hacked the stdlib so I'm not sure at all, but I wonder, is `eval` really needed?

Also, what would
```scheme
!(assert (empty))
```
do?  Would it raise an error, or silently output `[]`?  I believe it should raise an error, but I'm not sure your code behaves like that.  You should add a unit test for it.  Which is another question (to @vsbogd): where should the unit tests of the stdlib be?